### PR TITLE
Increase Http2 Header limit size to the MaxFrameSize

### PR DESCRIPTION
**Pull request by jkotalik**
Part of https://github.com/aspnet/AspNetCore/issues/12213

Increased the limit to the default max frame size.

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]

**Comment by jkotalik** [comment]
I don't believe this address all issues. The Http2Stream still calls ResetAndAbort with a protocol violation here: https://github.com/aspnet/AspNetCore/blob/caa910ceeba5f2b2c02c47a23ead0ca31caea6f0/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs#L207. I believe this is the real issue the stress app is hitting.

**Comment by halter73** [comment]
I agree that ProtocolError seems wrong for going over a server-configured limit that's not configured as part of the protocol. What do we think is better than Http2ErrorCode.PROTOCOL_ERROR though? My feeling is that Http2ErrorCode.INTERNAL is probably best.

**Review by halter73 — APPROVED** [review]

File: src/Servers/Kestrel/Core/src/Http2Limits.cs [hunk 1]
```diff
@@ -81,7 +81,7 @@ public int MaxFrameSize
         /// <summary>
         /// Indicates the size of the maximum allowed size of a request header field sequence. This limit applies to both name and value sequences in their compressed and uncompressed representations.
         /// <para>
-        /// Value must be greater than 0, defaults to 8192
+        /// Value must be greater than 0, defaults to 2^14 (16,384)
```
Comment (halter73) [inline src/Servers/Kestrel/Core/src/Http2Limits.cs]: Nit: I think it's OK just to use 16,384. People who care whether or not it's a power of two will intuitively know that it is.

**Comment by jkotalik** [comment]
I think this PR is fine for now, but I'll keep the original issue open to address the request line issues.

**Comment by Tratcher** [comment]
We should work out how we can send a 4XX response before resetting.

**Comment by halter73** [comment]
I think if you RST a connection no response at all is better than a 4XX. That aligns better with an HTTP/1.1 connection reset. We wont be able to write a 4XX anyway if the app has already started writing a response.

**Comment by Tratcher** [comment]
These are h2 stream resets, not tcp resets. They also happen as the request is received and prevent it from being dispatched to the app. Sending a 4XX would align with how we handle this class of errors in HTTP/1.1.

**Comment by halter73** [comment]
I think I understand what you're saying. I was thinking of receiving a HTTP/2 stream reset, not a tcp reset, but you're right I lost some of the context. Sending a 4XX for too-large headers is the right thing to do.

**Pull request merged**